lib/src/rewrite.rs

```diff
@@ -184,6 +184,13 @@ impl<'repo> CommitRewriter<'repo> {
         self.new_parents.retain(|parent| head_set.contains(parent));
     }
 
+    /// Records the old commit as abandoned with the new parents.
+    pub fn record_abandoned_commit(self) {
```

Comment:
nit: just call it `abandon()`?

I wouldn't expect `record_*()` is a consuming function, and it's similar to `.rebase()`.